config/jobs/kube-oidc-proxy/kube-oidc-proxy-presubmits.yaml

```diff
@@ -287,6 +287,102 @@ presubmits:
   - name: pull-kube-oidc-proxy-e2e-v1-16
     context: pull-kube-oidc-proxy-e2e-v1-16
     # Match everything except PRs that only touch docs/
+    always_run: false
+    cluster: gke
+    optional: false
+    max_concurrency: 4
+    agent: kubernetes
+    decorate: true
+    branches:
+    - master
+    labels:
+      preset-dind-enabled: "true"
+    spec:
+      containers:
+      - image: eu.gcr.io/jetstack-build-infra-images/golang-dind:20191205-e920769-1.13.4
+        args:
+        - runner
+        - make
+        - e2e
+        env:
+        - name: KUBE_OIDC_PROXY_K8S_VERSION
+          value: "1.16.4"
+        resources:
+          requests:
+            cpu: 6
+            memory: 12Gi
+        securityContext:
+          privileged: true
+          capabilities:
+            add: ["SYS_ADMIN"]
+        volumeMounts:
+        - mountPath: /lib/modules
+          name: modules
+          readOnly: true
+        - mountPath: /sys/fs/cgroup
+          name: cgroup
+      volumes:
+      - name: modules
+        hostPath:
+          path: /lib/modules
+          type: Directory
+      - name: cgroup
+        hostPath:
+          path: /sys/fs/cgroup
+          type: Directory
+
+  # kind based kube-oidc-proxy e2e job
+  - name: pull-kube-oidc-proxy-e2e-v1-17
+    context: pull-kube-oidc-proxy-e2e-v1-17
+    # Match everything except PRs that only touch docs/
+    always_run: true
```

Comment:
do you need to run both 1.17 and 1.18 jobs in presubmits? could we create periodic jobs for _all_ older k8s versions and have them displayed on testgrid with alerts for when things start failing instead?

Reply:
I will remove 17 from being run every time but I don't think it's too necessary to run periodic jobs imo

Reply:
sure :) how do you currently ensure that the proxy works okay against all supported versions of k8s ahead of a release?

Reply:
I actually forgot about this tbh, and will open up a no-op PR to check all versions